tests/httpwrappers/tests.py

```diff
@@ -491,13 +492,54 @@ def test_unsafe_redirect(self):
             'data:text/html,<script>window.alert("xss")</script>',
             "mailto:test@example.com",
             "file:///etc/passwd",
-            "é" * (MAX_URL_LENGTH + 1),
+            "é" * (MAX_URL_REDIRECT_LENGTH + 1),
         ]
-        for url in bad_urls:
-            with self.assertRaises(DisallowedRedirect):
-                HttpResponseRedirect(url)
-            with self.assertRaises(DisallowedRedirect):
-                HttpResponsePermanentRedirect(url)
+        for url, response_class in itertools.product(
+            bad_urls, (HttpResponseRedirect, HttpResponsePermanentRedirect)
```

Comment:
I leave it to you whether to keep all these duplicate tests for the Permanent flavor. I don't know if generally we duplicate all tests for it. I don't think we have to keep them throughout. We could keep this one we already touching? 🤔